File: Go/ExampleTests/common/replacepost_string.go
package common

type ReplacePostString struct {
	Id string
	UserId string
	Title string
	Body string
}

func NewReplacePostStringFromSlice(v []string) ReplacePostString {
	s := ReplacePostString{}
	if len(v) > 0 { s.Id = v[0] }
	if len(v) > 1 { s.UserId = v[1] }
	if len(v) > 2 { s.Title = v[2] }
	if len(v) > 3 { s.Body = v[3] }
	return s
}

// NewReplacePostStringFromText builds from the text form, e.g. Money as "25 USD".
func NewReplacePostStringFromText(text string) ReplacePostString {
	parts := RequireTokens(text, 4, "ReplacePost")
	return ReplacePostString{
		Id: parts[0],
		UserId: parts[1],
		Title: parts[2],
		Body: parts[3],
	}
}

func (s ReplacePostString) String() string {
	return Token(s.Id) + " " + Token(s.UserId) + " " + Token(s.Title) + " " + Token(s.Body)
}

func (s ReplacePostString) Equals(o ReplacePostString) bool {
	return DNCEqual(s.Id, o.Id) &&
		DNCEqual(s.UserId, o.UserId) &&
		DNCEqual(s.Title, o.Title) &&
		DNCEqual(s.Body, o.Body)
}

func EqualReplacePostStringSlices(a, b []ReplacePostString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/replacepost_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct ReplacePostString {
    pub id: String,
    pub userid: String,
    pub title: String,
    pub body: String,
}

impl ReplacePostString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            id: v.get(0).copied().unwrap_or("").to_string(),
            userid: v.get(1).copied().unwrap_or("").to_string(),
            title: v.get(2).copied().unwrap_or("").to_string(),
            body: v.get(3).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 4, "ReplacePost");
        Self {
            id: parts[0].clone(),
            userid: parts[1].clone(),
            title: parts[2].clone(),
            body: parts[3].clone(),
        }
    }
}

impl std::fmt::Display for ReplacePostString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.id), crate::common::tokens::token(&self.userid), crate::common::tokens::token(&self.title), crate::common::tokens::token(&self.body)].join(" "))
    }
}

impl PartialEq for ReplacePostString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.id, &other.id)
            && crate::common::dnc_equal(&self.userid, &other.userid)
            && crate::common::dnc_equal(&self.title, &other.title)
            && crate::common::dnc_equal(&self.body, &other.body)
    }
}
